### PR TITLE
Imu experiments

### PR DESCRIPTION
**Pull request by ShaneRozenLevy**
- Added read of imu on pi3hat at the robot and control loop level
- Enabled setting imu update frequency and mounting configuration
- I also had to fix the bug with the soft start max torque not being used in the robot contructor

### CODE REVIEW

File: include/kodlab_mjbots_sdk/mjbots_robot_interface.h [hunk 1]
```diff
@@ -124,6 +134,7 @@ class MjbotsRobotInterface {
   std::future<::mjbots::moteus::Pi3HatMoteusInterface::Output> can_result_;      /// future can result, used to check if
   /// response is ready
   SoftStart soft_start_;                                                      /// Soft Start object
+  ::mjbots::pi3hat::Attitude attitude_;                                       /// Robot attitude
```
Comment (jdcaporale) [inline include/kodlab_mjbots_sdk/mjbots_robot_interface.h:137]: I think if you are going to use smart pointers, they should be smart pointers everywhere including in the owner. Otherwise we are defeating the point of them.

**Review by jdcaporale — COMMENTED** [review]
Likely needs some documentation, but looks good other than the smart pointer missing. 
Ethan has built out a class version of what you did based on this branch, I don't know if we want to push this first, or just wait til his is ready in the next couple days.

File: include/kodlab_mjbots_sdk/pi3hat_moteus_interface.h [hunk 2]
```diff
@@ -203,7 +211,7 @@ class Pi3HatMoteusInterface : public kodlab::AbstractRealtimeObject{
     input.tx_can = { tx_can_.data(), tx_can_.size() };
     input.rx_can = { rx_can_.data(), rx_can_.size() };
     input.request_attitude = true;
-    input.attitude = &attitude_;
+    input.attitude = attitude_.get();
```
Comment (ShaneRozenLevy) [inline include/kodlab_mjbots_sdk/pi3hat_moteus_interface.h:214]: @jdcaporale  So right now nothing at the pi3hat level uses smart pointers. I think its fine to keep it that way, or it should be a separate PR

File: include/kodlab_mjbots_sdk/mjbots_robot_interface.h (hunk 1 above)
Comment (ShaneRozenLevy) [reply include/kodlab_mjbots_sdk/mjbots_robot_interface.h:137]: So its important that this is a copy of the pi3hat::Attitude object that is passed into the pi3hat moteus interface other wise you get a race condition. For this object I don't ever use it as a smart ptr.

**Comment by ShaneRozenLevy** [comment]
I'm not in a spot to finish up this PR, so I'm happy with either option in terms of getting this across the finish line.

In terms of what makes an easy to review PR I would recommend merging this one in, then Ethan's to keep everything small.
